ui/pages/permissions-connect/connect-page/connect-page.tsx

```diff
@@ -457,6 +475,42 @@ export const ConnectPage: React.FC<ConnectPageProps> = ({
                   </ButtonLink>
                 </Box>
               )}
+              {promptToCreateSolanaAccount && !solanaAccountCreated && (
```

Comment:
```suggestion
              {promptToCreateSolanaAccount && !existsSolanaAccount && (
```

Reply:
Updated to `solanaAccountExistsInWallet` here: https://github.com/MetaMask/metamask-extension/pull/31781/commits/b53b22ee44be827a8daa7bc2174448a45133b460